avocado/utils/genio.py

```diff
@@ -174,3 +174,22 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def safe_write_file(filename, data):
+    """
+    Write data to a file.
+
+    :param filename: Path to the file.
+    :type filename: str
+    :param data: data to be written.
+    :type data: str
+    """
+    try:
+        fd = os.open(filename, os.O_WRONLY)
+        ret = os.write(fd, data)
+    except Exception, error:
```

Comment:
Please don't use naked Exceptions. The `os.write` and `os.open` should only raise `OSError`.